core/rawdb/accessors_chain.go

```diff
@@ -695,33 +722,33 @@ func DeletePhCache(db ethdb.KeyValueWriter) {
 	}
 }
 
-// ReadCurrentPendingHeaderHash retreive's the heads hashes of the blockchain.
-func ReadCurrentPendingHeaderHash(db ethdb.Reader) common.Hash {
+// ReadBestPhKeyObject retreive's the bestPhKeyObject of the blockchain
+func ReadBestPhKeyObject(db ethdb.Reader) types.BestPhKeyObject {
 	data, _ := db.Get(phHeadKey)
 	// get the ph cache keys.
 	if len(data) == 0 {
-		return common.Hash{}
+		return types.BestPhKeyObject{}
 	}
-	hash := common.Hash{}
-	if err := rlp.DecodeBytes(data, &hash); err != nil {
-		return common.Hash{}
+	bestPhKey := new(types.BestPhKeyObject)
+	if err := rlp.DecodeBytes(data, bestPhKey); err != nil {
+		return types.BestPhKeyObject{}
```

Comment:
If there's an error wouldn't it be best to either return the error (or if following a pattern), return nil instead of a new object?

Reply:
bestPhKey is converted back to the common.Hash